### ISSUE TITLE
[ENHANCEMENT] Support untagged VLAN for storage network

### ISSUE THREAD

**Issue by starbops** [harvester/harvester#4323]
**Is your enhancement related to a problem? Please describe.**

The storage network configuration only supports tagged VLAN currently. The "VLAN ID" field on the UI is mandatory. Users might want to set up their storage network on an untagged VLAN network. This is not achievable now.

**Describe the solution you'd like**

For such use cases, users should be able to associate the storage network with a cluster network without specifying the VLAN ID.

**Describe alternatives you've considered**

**Additional context**

Original discussion thread: https://rancher-users.slack.com/archives/C01GKHKAG0K/p1690380258466139

We already supported [untagged VM networks](https://docs.harvesterhci.io/v1.1/networking/harvester-network#untagged-network); such capability is also needed for the storage network.

**Comment by andyrue** [comment]
I ran into this issue while testing things out on a Proxmox host. By default Proxmox NICs don't allow passing of tagged traffic, and I was using a dedicated NIC and Switch, so I didn't need VLANs. So if anybody runs into this, it may seem obvious, but make sure your infrastructure is configured to handle whatever VLAN ID you give it, even if you don't need VLANs.

**Comment by w13915984028** [comment]
@andyrue 

This is an interesting scenario, could you provide more context, thanks:

When Equinix supports machine(Harvester Node) to have mutli NICs and all those NICs are in plain network, how are they differentiated? Suppose each NIC is put in a separate subnet, then:

(1) Does Equinix allow machine to set IP itself  or machine should use the pre-allocated/DHCP based IP from equinix?

(2) How does Equinix price the IP?  The `StorageNetwork` needs a (big) subnet to work, this will be an additional cost to user if every IP costs.

The `StorageNetwork` adds a second IP to most of the Longhorn related PODs for dedicated storage usage, the packets go through the provider (e.g. Equinix infrastructure) network, but do not transfer the L3 subnet, and do not communicate with the external.

At the moment, the `StorageNetwork` depends on the VLAN Network for network isolation and IP management. If user does not have a pure L3 network infrastructure (indeed, most do not have), enabling the `StorageNetwork` on untagged network is error-prone.

Supporting the pure L3 network will be combined with the in-plan new feature https://github.com/harvester/harvester/issues/2322.

**Comment by dfaltum** [comment]
This isn't just a GUI issue. The validator webhook rejects omitting this field: https://github.com/harvester/harvester/blob/c965b66d5d3bddecbe9fced8b760fd03dbae2d4c/pkg/webhook/resources/setting/validator.go#L1294-L1299

```
# settings.harvesterhci.io "storage-network" was not valid:
# * value: The valid value range for VLAN IDs is 1 to 4094
```

**Comment by rrajendran17** [comment]
@dfaltum yes changes will be needed both in harvester and GUI if we are going to support this.

**Comment by rrajendran17** [comment]
Summarizing the discussion from slack on supporting untagged network for storage networks.

1.The functionality of untagged vlan in harvester is similar to that of vlan with tag 1.
   The packets are sent out of harvester node as untagged in case of vlan with ID 1 and ip range allocated from dhcp pool is same
   for vlan 1 and untagged,

2.Using/supporting of untagged network is more of user experience who does not need to configure any vlans.
   At any point of time, vlan 1 could be used when used with vlan unaware external switches.

3.Using untagged network with storage networks could be supported with the following,
  a.Using vlan 1 or untagged in mgmt cluster(vlan 1 or untagged) should be restricted/avoided as ip range allocated will be same
    for storage network and it breaks the purpose of using storage networks.

 b. Using vlan 1 or untagged could be supported with separate cluster network 

c.Using vlan 1 or untagged could be supported with mgmt network in different vlan.

d.storage network allocates ip addresses to LH pods running on all the nodes in the cluster in the same subnet with the vlan id specified usually.The communication happens between the LH pods in different nodes is pure L2.
In case of untagged networks, this is still possible but with the external switch acting as L2 in vlan unaware mode.

e.User could select untagged mode based on their network configuration keeping in mind that storage network must use isolated network and it must be within same L2.

Changes required:

1.GUI changes to support untagged option 
2.Harvester webhook changes to modify vlan range checks and support vlan id with 0
3.Harvester webhook changes to reject vlan same as mgmt cluster if configured under mgmt cluster

cc @w13915984028 @starbops @mingshuoqiu 
Please let me know your views.

**Comment by w13915984028** [comment]
@rrajendran17 

Thanks for the summary. I have no more concerns.

**Comment by rrajendran17** [comment]
As mentioned above there is a case where user might configure untagged with mgmt cluster and mgmt interface has no vlan configured.This applies to vlan with id 1 as well.

Currently there is no easy way getting the vlan of mgmt interface in harvester code for validating this.Is it worth adding a webhook change to prevent users configuring untagged vlan for storage network in mgmt cluster with mgmt interface with no vlan? Or can we add a tooltip or documentation to users, so that users should know the right way to configure storage network so that its used effectively ?

IMO, adding information to the doc is better as all network configurations need not to be validated at webhook.
@w13915984028 @starbops @mingshuoqiu any thoughts about this ?

**Comment by starbops** [comment]
Yeah, I think addressing it with a tooltip and providing a deeper explanation in the document is enough.

**Comment by rrajendran17** [comment]
The backend changes are ready and backported to v1.5.
Not moving this to "Ready to Testing" until front end https://github.com/harvester/harvester/issues/6501 changes are ready.

**Comment by noahgildersleeve** [comment]
Tested in Harvester `v1.5.0-rc1` and Rancher `v2.11-0ea48604614e293227c22ccf399b7a3eeea547fc-head` with `ui-extensions` `v1.5-head`. Validated that untagged VLAN worked with storage network when setting via Harvester UI and via UI in upstream Rancher cluster with Harvester imported.

![Image](https://github.com/user-attachments/assets/1909c38d-c177-4baf-bbdc-a9b8f4e10f8f)
![Image](https://github.com/user-attachments/assets/53b67e32-bcc1-4169-b4e6-056246751459)
